File: Java/58. Length of Last Word.java
/**
 * https://leetcode.com/problems/length-of-last-word/
 */

class Solution {
    // O(n) time complexity
    // O(1) space complexity
    public int lengthOfLastWord(String s) {
        int res = 0;
        for (int j = s.length() - 1; j >= 0; j--) {
            char c = s.charAt(j);
            if (Character.isLetter(c)) {
                res++;
            } else if (res > 0) {
                return res;
            }
        }
        return res;
    }
}

File: Kotlin/58. Length of Last Word.kt
/**
 * https://leetcode.com/problems/length-of-last-word/
 */

class Solution {
    // O(n) time complexity
    // O(1) space complexity
    fun lengthOfLastWord(s: String): Int {
        var res = 0
        for (j in s.length - 1 downTo 0) {
            val c = s[j]
            if (c != ' ') {
                res++
            } else if (res > 0)
                return res
        }
        return res
    }
}